File: AndroidBasic/app/src/main/java/com/ylg/androidbasic/MainActivity.java
package com.ylg.androidbasic;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.OrientationHelper;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;

/**
 * https://www.w3cschool.cn/uawnhh/qbe84ozt.html
 * http://c.biancheng.net/view/2918.html
 */
public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;
    private LinearLayoutManager linearLayoutManager;

    String TAG = MainActivity.class.getName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerview);
        //设置固定大小
        //recyclerView.setHasFixedSize(true);

        linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.VERTICAL);
        recyclerView.setLayoutManager(linearLayoutManager);

        adapter = new RecyclerAdapter(this);
        recyclerView.setAdapter(adapter);

    }
}

File: MyKotlinStudy/app/src/main/java/com/ylg/mykotlinstudy/MainActivity.kt
package com.ylg.mykotlinstudy

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {

    private var recyclerView: RecyclerView? = null
    private var adapter: RecyclerAdapter? = null
    private var linearLayoutManager: LinearLayoutManager? = null

    var TAG = MainActivity::class.java.name

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerview)
        //设置固定大小
        //recyclerView.setHasFixedSize(true);

        linearLayoutManager = LinearLayoutManager(this)
        linearLayoutManager?.orientation = RecyclerView.VERTICAL
        recyclerView?.layoutManager = linearLayoutManager

        adapter = RecyclerAdapter(this)
        //recyclerView?.adapter = adapter
        recyclerView?.setAdapter(adapter)

    }
}